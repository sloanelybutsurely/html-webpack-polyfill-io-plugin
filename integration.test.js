'use strict'

const fs = require('fs')
const path = require('path')
const rimraf = require('rimraf')
const webpack = require('webpack')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const HtmlWebpackPolyfillIOPlugin = require('./')

const OUTPUT_DIR = path.join(__dirname, 'dist')

const compile = (config, cb) =>
  webpack(
    {
      entry: path.join(__dirname, 'fixtures', 'index.js'),
      output: {
        path: OUTPUT_DIR,
      },
      plugins: [
        new HtmlWebpackPlugin(),
        new HtmlWebpackPolyfillIOPlugin(config),
      ],
    },
    err => {
      expect(err).toBeFalsy()
      const html = fs
        .readFileSync(path.join(OUTPUT_DIR, 'index.html'))
        .toString()
      cb(html)
    }
  )

describe('HtmlWebpackPolyfillIOPlugin', () => {
  beforeEach(done => {
    rimraf(OUTPUT_DIR, done)
  })
  describe('simple', () => {
    it('adds the polyfill script to the head', done => {
      compile(undefined, html => {
        expect(html).toMatchSnapshot()
        done()
      })
    })
  })
  describe('with configuration', () => {
    it('adds the polyfill script to the head with the configuration', done => {
      compile(
        {
          minify: true,
          features: ['default-3.6'],
          flags: 'gated',
          rum: true,
        },
        html => {
          expect(html).toMatchSnapshot()
          done()
        }
      )
    })
  })
  describe('with callback', () => {
    it('adds the polyfill script to the body and a resource hint to the head', done => {
      compile(
        {
          callback: 'ready',
          minify: true,
          features: ['default-3.6'],
          flags: 'gated',
          rum: true,
        },
        html => {
          expect(html).toMatchSnapshot()
          done()
        }
      )
    })
  })
})
