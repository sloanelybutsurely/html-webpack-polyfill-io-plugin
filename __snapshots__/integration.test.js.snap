// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlWebpackPolyfillIOPlugin simple adds the polyfill script to the head 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>Webpack App</title>
  <script src=\\"https://cdn.polyfill.io/v2/polyfill.js\\"></script></head>
  <body>
  <script type=\\"text/javascript\\" src=\\"main.js\\"></script></body>
</html>"
`;

exports[`HtmlWebpackPolyfillIOPlugin with callback adds the polyfill script to the body and a resource hint to the head 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>Webpack App</title>
  <link href=\\"https://cdn.polyfill.io/v2/polyfill.min.js?features=default-3.6&flags=gated&callback=ready&rum=1\\" type=\\"preload\\" as=\\"script\\"></head>
  <body>
  <script src=\\"https://cdn.polyfill.io/v2/polyfill.min.js?features=default-3.6&flags=gated&callback=ready&rum=1\\" async></script><script type=\\"text/javascript\\" src=\\"main.js\\"></script></body>
</html>"
`;

exports[`HtmlWebpackPolyfillIOPlugin with configuration adds the polyfill script to the head with the configuration 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"UTF-8\\">
    <title>Webpack App</title>
  <script src=\\"https://cdn.polyfill.io/v2/polyfill.min.js?features=default-3.6&flags=gated&rum=1\\"></script></head>
  <body>
  <script type=\\"text/javascript\\" src=\\"main.js\\"></script></body>
</html>"
`;
